src/utilities/use-deep-callback.tsx

```diff
@@ -2,6 +2,25 @@ import {useCallback} from 'react';
 import {EffectCallback, DependencyList, Comparator} from '../types';
 import {useDeepCompareRef} from './use-deep-compare-ref';
 
+/**
+ * A replacement for React.useCallback that'll allow for custom and deep compares.
+ * @see {@link https://reactjs.org/docs/hooks-reference.html#usecallback}
+ * @param callback Accepts a callback that's forwarded to React.useCallback
+ * @param dependencies A dependency array similar to React.useCallback however it utilizes a deep compare
+ * @param customCompare Opportunity to provide a custom compare function
+ * @returns A memoized callback
+ * @example
+ * function ComponentExample() {
+ *  const [, forceUpdate] = useState();
+ *  const obj = {a: 1};
+ *
+ *  const handleClick = useDeepCallback(() => {
+ *    forceUpdate(obj);
+ *  }, [obj]);
```

Comment:
I'm not sure I understand what this hook is doing. Would `forceUpdate` only be called once?

Reply:
Added a new example to hopefully make this clearer 😄